subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -197,6 +204,10 @@ public void handleLostState() {
             LOGGER.warn("Spent {} ms registering watches for file system events", endTime);
             // TODO: Move start watching early enough so that the root is always empty
             return currentRoot.empty();
+        } catch (InsufficientResourcesForWatchingException ex) {
+            logInsufficientResourcesException(ex);
+            closeUnderLock();
+            return currentRoot.empty();
```

Comment:
Would it make sense to do this as part of the `Exception` catch-block instead?

Reply:
There is now only one catch block and one logging handling in `WatchingVirtualFileSystem`.